core/field_matrix_color.js

```diff
@@ -92,14 +92,438 @@ Blockly.FieldMatrixColour.prototype.render_ = function () {
  * @return {Array.<Object>} colors Array of colors
  */
 Blockly.FieldMatrixColour.prototype.defaultColors = function () {
-  let colorArray = []
-  for (let i = 0; i <= 48; i++) {
-    let temp = Blockly.FieldMatrixColour.DEFAULT_COLOR
-    if (i === 0 || i === 6 || i === 42 || i === 48) {
-      temp = ''
+  const colorArray = [
+    {
```

Comment:
Can you put in the JS doc what each property is supposed to be? Not sure what xRange yRange are? Id's seem to be tied to the position but not sure.

Reply:
Added